examples/check_monotonicity.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python
+# Copyright (c) 2015, Ecole Polytechnique Federale de Lausanne, Blue Brain Project
+# All rights reserved.
+#
+# This file is part of NeuroM <https://github.com/BlueBrain/NeuroM>
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+#     1. Redistributions of source code must retain the above copyright
+#        notice, this list of conditions and the following disclaimer.
+#     2. Redistributions in binary form must reproduce the above copyright
+#        notice, this list of conditions and the following disclaimer in the
+#        documentation and/or other materials provided with the distribution.
+#     3. Neither the name of the copyright holder nor the names of
+#        its contributors may be used to endorse or promote products
+#        derived from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY
+# DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+# ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+
+'''Check if neuron diameters are monotonic'''
+
+from neurom.core.tree import ipreorder
+from neurom.core.dataformat import COLS
+
+
+def is_monotonic(tree, tol):
+    '''Check if tree is monotonic, that is if each child has smaller or
+        equal diameters from its parent
+
+        Input
+
+            tree : tree object
+
+        tol
+            numerical precision
+    '''
+
+    for node in ipreorder(tree):
+
+        if node.parent is not None:
+
+            if node.value[COLS.R] > node.parent.value[COLS.R] + tol:
+
+                return False
+
+    return True
+
+
+def check_monotonicity(neuron, tol=1e-6):
+    '''Check if neuron's dendrites have monotonic diameters
+
+        Input
+            neuron : neuron object
+
+        tol
+            numerical precision
+    '''
+
+    for i, neurite in enumerate(neuron.neurites):
+
+        print '{0} {1:30} {2}'.format(i, neurite.type, is_monotonic(neurite, tol))
```

Comment:
I know this is an example, but it would make sense to make it into a checker app, or part of the morph_check app at some point. So better to get used to using loggers and log levels for output.